data_structures/linked_list/circular_doubly_linked_list.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * @file
+ *
+ * @details
+ Circular Doubly Linked List combines the properties of a doubly linked list and a circular linked list in which two consecutive elements are linked or connected by the previous and next pointer and the last node points to the first node via the next pointer and the first node points last node via the previous pointer.
+
+ In this implementation, the menu driven code is written along with functions to insert at start, insrt at last, delete first node, delete last node and display list functions are coded.
+ 
+ * Author: [Sahil Kandhare](https://github.com/SahilK-027)
+ * Created on 12/10/2022
+ *
+ *
+ * Cicular doubly linked list in C
+ *
+ * Compiled in Visual Studio Code
+ *
+ */
+
+#include <stdio.h>     /// for IO operations
+#include <stdlib.h>    /// for including functions involving memory allocation such as `malloc`
+#include <inttypes.h>  /// to provide a set of integer types with universally consistent definitions that are operating system-independent
+/**
+ * @brief Node Structure Cicular Doubly linked list.
+ */
+struct node
+{
+    struct node *prev, *next;         ///< List pointers
+    int64_t value;                   ///< Data stored on each node
+};
+
+/**
+ * @brief   Insert the node at start of list
+ * @param   start   starting pointer of list
+ * @param   data    the data which we want to insert into list
+ * @returns void
+ */
+void insert_at_start(struct node **start, int data)
+{
+    struct node *n1;
+    n1 = (struct node *)malloc(sizeof(struct node));
+    n1->value = data;
+    if (*start == NULL)
+    {
+        n1->next = n1;
+        n1->prev = n1;
+        *start = n1;
+    }
+    else
+    {
+        struct node *temp;
+        temp = (*start)->prev;
+        n1->next = *start;
+        (*start)->prev = n1;
+        n1->prev = temp;
+        temp->next = n1;
+        *start = n1;
+    }
+    printf("\nNode succesfully inserted\n\n");
+}
+
+/**
+ * @brief   Insert the node at end of list
+ * @param   start   starting pointer of list
+ * @param   data    the data which we want to insert into list
+ * @returns void
+ */
+void insert_at_last(struct node **start, int data)
+{
+    struct node *n1;
+    n1 = (struct node *)malloc(sizeof(struct node));
+    n1->value = data;
+    if (*start == NULL)
+    {
+        n1->next = n1;
+        n1->prev = n1;
+        *start = n1;
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = (*start)->prev;
+        n1->prev = temp2;
+        n1->next = temp1;
+        temp1->prev = n1;
+        temp2->next = n1;
+    }
+    printf("\nNode succesfully inserted\n\n");
+}
+
+/**
+ * @brief   Deletion of the first node in list function
+ * @param   start   starting pointer of list
+ * @returns void
+ */
+void delete_fist_node(struct node **start)
```

Comment:
Spelling error here, I think.
```suggestion
void delete_first_node(struct node **start)
```

Reply:
Yes